contracts/src/Assets.sol

```diff
@@ -159,6 +187,42 @@ library Assets {
         emit IGateway.TokenSent(token, sender, destinationChain, destinationAddress, amount);
     }
 
+    // @dev Transfer Polkadot-native tokens back to Polkadot
+    function _sendForeignToken(
+        bytes32 foreignID,
+        address token,
+        address sender,
+        ParaID destinationChain,
+        MultiAddress calldata destinationAddress,
+        uint128 destinationChainFee,
+        uint128 amount
+    ) internal returns (Ticket memory ticket) {
+        if (destinationChainFee == 0) {
+            revert InvalidDestinationFee();
+        }
+
+        Token(token).burn(sender, amount);
+
+        ticket.dest = destinationChain;
+        ticket.costs = _sendForeignTokenCosts(destinationChainFee);
+
+        if (destinationAddress.isAddress32()) {
+            // The receiver has a 32-byte account ID
+            ticket.payload = SubstrateTypes.SendForeignTokenToAddress32(
+                foreignID, destinationChain, destinationAddress.asAddress32(), destinationChainFee, amount
+            );
+        } else if (destinationAddress.isAddress20()) {
+            // The receiver has a 20-byte account ID
+            ticket.payload = SubstrateTypes.SendForeignTokenToAddress20(
+                foreignID, destinationChain, destinationAddress.asAddress20(), destinationChainFee, amount
+            );
+        } else {
+            revert Unsupported();
+        }
```

Comment:
Need to double check the execution cost on AH(i.e. `assetHubReserveTransferFee`) is included.

Reply:
For now I'm assuming `assetHubReserveTransferFee` (which is configured to have considerable buffer) will cover costs for xcm execution on AH for both ENA and PNA flows.

We should probably rename the field to something like `assetHubXcmFee`